DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/searchSortFilterAndPaginate.ts

```diff
@@ -213,7 +214,59 @@ export class SearchSortFilterAndPaginate {
     return newResultCountMessage;
   }
 
-  private static scrollToTop() : void {
+  private static scrollToTop(): void {
     window.scrollTo(0, 0);
   }
+
+  private updatePageNumber(pageNumber: number, searchableData: ISearchableData): void {
+    this.page = pageNumber;
+
+    SearchSortFilterAndPaginate.fixPageNumber();
+    const currentPath = window.location.pathname;
+    const urlParts = currentPath.split('/');
+    const newUrl = `${urlParts.slice(0, -1).join('/')}/${pageNumber.toString()}`;
+    window.history.replaceState({}, '', newUrl);
+
+    this.updateSearchableElementLinks(searchableData);
+  }
+
+  private static getPageNumber(): number {
+    SearchSortFilterAndPaginate.fixPageNumber();
+    const currentPath = window.location.pathname;
+    const urlParts = currentPath.split('/');
+    return parseInt(urlParts[urlParts.length - 1], 10);
+  }
+
+  /* Guarantees the last element of the path is a number with no trailing slashes */
+  private static fixPageNumber(): void {
+    const currentPath = window.location.pathname;
+    const urlParts = currentPath.split('/');
+    if (urlParts[urlParts.length - 1] === '') {
+      urlParts.pop();
+    }
+
+    const pageNumber = parseInt(urlParts[urlParts.length - 1], 10);
+
+    if (Number.isNaN(pageNumber)) {
+      const newUrl = `${urlParts.join('/')}/1`;
+      window.history.replaceState({}, '', newUrl);
+    }
+  }
+
+  private updateSearchableElementLinks(searchableData: ISearchableData): void {
+    const setReturnPage = !this.searchEnabled || getQuery().length === 0;

```

Comment:
Good point, it doesn't. I'm going to add `.trim()` to the return value of `getQuery()` so that it can be used more intuitively, rather than changing this here.